pallets/schemas/src/benchmarking.rs

```diff
@@ -339,6 +376,256 @@ mod benchmarks {
+		let intent_ids = generate_intents::<T>(m.try_into().expect("should convert"));
+
+		#[extrinsic_call]
+		create_intent_group(RawOrigin::Signed(sender), bounded_name, intent_ids);
+
+		ensure!(
+			CurrentIntentGroupIdentifierMaximum::<T>::get() > 0,
+			"Created intent group count should be > 0"
+		);
+		ensure!(IntentGroups::<T>::get(1).is_some(), "Created intent group should exist");
+		Ok(())
+	}
+
+	#[benchmark]
+	fn create_intent_group_via_governance(
+		m: Linear<0, { T::MaxSchemaSettingsPerSchema::get() }>,
+	) -> Result<(), BenchmarkError> {
+		let sender: T::AccountId = whitelisted_caller();
+		let namespace = vec![b'a'; PROTOCOL_NAME_MIN as usize];
+		let descriptor = vec![b'b'; DESCRIPTOR_MAX as usize];
+		let name: Vec<u8> = namespace
+			.into_iter()
+			.chain(vec![b'.'].into_iter())
+			.chain(descriptor.into_iter())
+			.collect();
+		let bounded_name = BoundedVec::try_from(name).expect("should resolve");
+		let intent_ids = generate_intents::<T>(m.try_into().expect("should convert"));
+
+		#[extrinsic_call]
+		create_intent_group_via_governance(
+			RawOrigin::Root,
+			sender.clone(),
+			bounded_name,
+			intent_ids,
+		);
+
+		ensure!(
+			CurrentIntentGroupIdentifierMaximum::<T>::get() > 0,
+			"Created intent group count should be > 0"
+		);
+		ensure!(IntentGroups::<T>::get(1).is_some(), "Created intent group should exist");
+		Ok(())
+	}
+
+	#[benchmark]
+	fn propose_to_create_intent_group() -> Result<(), BenchmarkError> {
+		let sender: T::AccountId = whitelisted_caller();
+		let namespace = vec![b'a'; PROTOCOL_NAME_MIN as usize];
+		let descriptor = vec![b'b'; DESCRIPTOR_MAX as usize];
+		let name: Vec<u8> = namespace
+			.into_iter()
+			.chain(vec![b'.'].into_iter())
+			.chain(descriptor.into_iter())
+			.collect();
+		let bounded_name = BoundedVec::try_from(name).expect("should convert");
+		let intent_ids = generate_intents::<T>(0);
+
+		#[extrinsic_call]
+		propose_to_create_intent_group(RawOrigin::Signed(sender), bounded_name, intent_ids);
+
+		assert_eq!(T::ProposalProvider::proposal_count(), 1);
+		Ok(())
+	}
+
+	#[benchmark]
+	fn update_intent_group(
+		m: Linear<0, { T::MaxIntentsPerIntentGroup::get() }>,
+	) -> Result<(), BenchmarkError> {
+		let sender: T::AccountId = whitelisted_caller();
+		let namespace = vec![b'a'; PROTOCOL_NAME_MIN as usize];
+		let descriptor = vec![b'b'; DESCRIPTOR_MAX as usize];
+		let name: Vec<u8> = namespace
+			.into_iter()
+			.chain(vec![b'.'].into_iter())
+			.chain(descriptor.into_iter())
+			.collect();
+		let bounded_name = BoundedVec::try_from(name).expect("should resolve");
+		let intent_ids = generate_intents::<T>(T::MaxIntentsPerIntentGroup::get() as usize);
+		let mut initial_intent_ids = intent_ids.clone();
+		initial_intent_ids.truncate(1);
+
+		let (intent_group_id, _) =
+			SchemasPallet::<T>::create_intent_group_for(bounded_name, initial_intent_ids)
+				.expect("should create intent group");
+
+		#[extrinsic_call]
+		update_intent_group(
```

Comment:
nit: if the name of benchmark and extrinsic is the same you can use `_` instead of the name